package fr.epita.factorial.tests;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.fail;

import org.junit.Assert;
import org.junit.Test;

import fr.epita.factorial.main.Factorial;

public class FactorialTest {
	
	@Test
	public void testCalculateFactorialOne()
	{
        assertEquals(1, Factorial.calculateFactorial(0));
	}

	@Test
	public void testCalculateFactorialFive()
	{
        assertEquals(120, Factorial.calculateFactorial(5));
	}
	
	@Test
	public void testNegativeFactorial()
	{
		Exception e;
		try
		{
			Factorial.calculateFactorial(-1);
		}
		catch(IllegalArgumentException iae)
		{
			e = iae;
		}
		Assert.assertNotNull(e);	
	}
}
